src/Pages/Category/Category.jsx

```diff
@@ -1,123 +1,45 @@
-
-    }, {
-
-        id: '4',
-        name: 'Grande',
-        descripton: 'Leather Coach Bag with adjustable starps.',
-        brand: 'Blossom Pouch',
-        price: {
-            current: '39.49',
-            actual: "78.66",
-            discountPercentage: '50'
-        },
-        rating: "4.6",
-        totoalRatings: '43',
-
-        imgUrl: '../Assets/laura-chouette-Atomp7YdMaE-unsplash 1.png',
-        fullDescription: " Lorem ipsum dolor sit amet consectetur adipisicing elit. Tenetur, magnam fuga harum ab aliquid facilis quo accusantium nemo reiciendis rem eaque obcaecati, culpa aliquam illum numquam blanditiis, inventore amet repellendus!"
-
-    }, {
-
-        id: '4',
-        name: 'Grande',
-        descripton: 'Leather Coach Bag with adjustable starps.',
-        brand: 'Blossom Pouch',
-        price: {
-            current: '39.49',
-            actual: "78.66",
-            discountPercentage: '50'
-        },
-        rating: "4.5",
-        totoalRatings: '43',
-
-        imgUrl: '../Assets/laura-chouette-Atomp7YdMaE-unsplash 1.png',
-        fullDescription: " Lorem ipsum dolor sit amet consectetur adipisicing elit. Tenetur, magnam fuga harum ab aliquid facilis quo accusantium nemo reiciendis rem eaque obcaecati, culpa aliquam illum numquam blanditiis, inventore amet repellendus!"
-
-    }
-];
-function Category({ targetURL }) {
-    const { category } = useParams();
+import { fetchData } from "../../Services/network";
+
+function Category() {
+    const { targetAPI } = useParams();
+    const [products, setProducts] = useState();
+
+    useEffect(() => {
+        const fetchDataAsync = async () => {
+            try {
+                const result = await fetchData(`https://store-osn9.onrender.com/products/${targetAPI}`);
+                setProducts(result);
+            } catch (error) {
+                console.error(error);
+            }
+        }
+        fetchDataAsync();
+    }, [])
 
     return (
-        <>
-            {console.log(targetURL)}
-            {console.log(category)}
-            <BreadCrumb MainTitle={"Home"} SubTitle={category} />
+        <div>
+            <BreadCrumb MainTitle={"Home"} SubTitle={targetAPI} />
             <Box sx={{ flexGrow: 1, paddingInline: "1rem", marginBlock: "2rem" }} >
                 <Grid container spacing={{ xs: 2, md: 3 }} columns={{ xs: 4, sm: 6, md: 12, lg: 15 }}>
-                    {products.map((product) => (
+                    {products? products.map((product) => (
                         <Grid item xs={2} sm={2} md={3} lg={3} key={product.id}>
                             <ProductCard product={product} isDetailed={true} />
                         </Grid>
-                    ))}
+                    )) : 'No data found'}
                 </Grid>
```

Comment:
can you handle errors and loading please?

Reply:
I merged it with main